packages/components/src/autocomplete/index.tsx

```diff
@@ -182,6 +182,8 @@ export function useAutocomplete( {
 				setAutocompleter( null );
 				setAutocompleterUI( null );
 				event.preventDefault();
+				// This prevents the block editor from handling the escape key to unselect the block.
+				event.stopPropagation();
```

Comment:
These remaining stopPropagation calls were necessary because if I don't do this the "clear block selection" shortcut triggers as well which causes the block to unselect and lose focus.

What I don't understand is why we didn't have these in place in "trunk" because even in trunk, it seems that the handler of the "unselect" shortcut should trigger because we're just hitting "Escape", I'm not sure why it's not called on trunk. It seemed logical for me that we add these when dialogs are open and we only want to close the dialogs.

Reply:
In that case we should check `event.defaultPrevented` for clearing selection. Did we forget that there?

Reply:
Ok, I think it's solved now :)